### PR TITLE
MNT: Modularize Rocket Draw

### PR DESCRIPTION
**Pull request by MateusStano**
## Pull request type

- [X] Code maintenance (refactoring, formatting, tests)

## Checklist

- [X] Lint (`black rocketpy/ tests/`) has passed locally 
- [X] All tests (`pytest tests -m slow --runslow`) have passed locally
- [X] `CHANGELOG.md` has been updated (if relevant)

## Current behavior

The method of `_RocketPlots` relating to the rocket drawing was too big and hard to change/improve/maintain

## New behavior

The `.draw()` method has been separated into several different methods. No code relating to the plot has been changed.

## Breaking change

- [X] No

## Additional information

I was trying to add sensors to the drawing and it was really hard to work with the drawing, so I refactored it.

A quick note:

- There are a few lines of repeated code that could still be turned into functions. These are usually some matplotlib one-liners. I did not create a function for these because it affected readability negatively.

### CODE REVIEW

**Review by Gui-FernandesBR — COMMENTED** [review]
Overall, very good PR. I understand you refactored the code without changing the behavior. You probably ran all the example notebooks to check the plots are being rendered correctly.

Some lines in the `_draw_nozzle_tube` function are missing coverage.
Could you please take a look at that? Just to ensure the code never brakes.

File: rocketpy/plots/rocket_plots.py [hunk 1]
```diff
@@ -225,113 +241,105 @@ def draw(self, vis_args=None):
...
+        ax.plot(
+            [x_nosecone[-1], x_nosecone[-1]],
+            [y_nosecone[-1], -y_nosecone[-1]],
+            color=vis_args["nose"],
+            linewidth=vis_args["line_width"],
+        )
+        # Add the nosecone to the list of drawn surfaces
+        drawn_surfaces.append(
+            (surface, x_nosecone[-1], surface.rocket_radius, x_nosecone[-1])
+        )
```
Comment (Gui-FernandesBR) [inline rocketpy/plots/rocket_plots.py:278]: This is strange to me. You add the surface to a list, which was passed as an argument to the function. If you don't return the list with the added surfaces, is this code line really necessary? I'm afraid that everytime you are starting with an empty list.

```
        # Add the nosecone to the list of drawn surfaces
        drawn_surfaces.append(
            (surface, x_nosecone[-1], surface.rocket_radius, x_nosecone[-1])
        )
```

Comment (Gui-FernandesBR) [reply rocketpy/plots/rocket_plots.py:278]: This behavior is also present in other parts of the code.

**Comment by Gui-FernandesBR** [comment]
Also, @MateusStano can you please test the implementation in a multi-stage rocket? I think the NDRT example notebook may help you with that.

File: rocketpy/plots/rocket_plots.py (hunk 1 above)
Comment (MateusStano) [reply rocketpy/plots/rocket_plots.py:278]: The `drawn_surfaces` list is created and then appended to the various draw methods I created. That same list is used later to draw the tubes later.

Remember that a list in Python is just a pointer. Changing the name of the list variable does not change the address that it points to.

**Comment by Gui-FernandesBR** [comment]
> > Also, @MateusStano can you please test the implementation in a multi-stage rocket? I think the NDRT example notebook may help you with that.
> 
> The drawing ends up identical to the way it was before. Did you find any problems there?

Nope. Sorry, I just did not have enough time to checkout your branch and run the examples.

If you have tested all the examples, I trust you. Go ahead and merge it.

**Review by Gui-FernandesBR — APPROVED** [review]
Good work.

**Pull request merged**